sdk/aqueduct/artifacts/create.py

```diff
@@ -111,4 +113,10 @@ def create_param_artifact(
         ],
     )
 
-    return to_artifact_class(dag, output_artifact_id, artifact_type, default)
+    if isinstance(default, LocalData):
```

Comment:
I'm really confused as to why we're not doing this in `client.create_param()`. And why we also have such an `isinstance()` check in `construct_param_spec()`. Let's extract the local data in one place and then feed it into `create_param_artifact` no?

Reply:
Yeah, I think putting it at` client.create_param()` save a lot of unnecessary type checking, there is no reason we should put it in `construct_param_spec` Should be fixed here!